### PR TITLE
remove bam compression

### PR DESCRIPTION
**Pull request by moonso**
This PR removes the code and tests for handling bam compression. The bam compression was a one of thing and when all bam files where compressed it will not be used again.

## Review
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [x] **MAJOR** - Some CLI commands have been removed

### CODE REVIEW

File: tests/meta/compress/test_compress_common.py [hunk 1]
```diff
@@ -60,24 +60,10 @@ def test_get_nlinks_three_links(project_dir):
     assert nlinks == 3
 
 
-def test_get_scout_case_no_scout_case(compress_api, case_id, caplog):
+def test_get_latest_version_no_housekeeper_files(compress_api, caplog):
     """test get_bam_dict method when there is not case in scout"""
```
Comment (henrikstranneheim) [inline tests/meta/compress/test_compress_common.py:64]: Looks like this docstring needs to be updated

**Review by henrikstranneheim — APPROVED** [review]
Great! Always good to remove code that is not needed anymore.
Is the bam compression processes in the crontab still commented out? If so we should open a PR to remove them and link these to PRs in the PR header

**Comment by moonso** [comment]
Hi, @patrikgrenfeldt and @henrikstranneheim . I would like to get this deployed before tonights run so could you please have a quick look?

**Comment by henrikstranneheim** [comment]
I am sorry - have to go get the kids now. If @patrikgrenfeldt do not have time I could do it first thing tomorrow.

**Comment by moonso** [comment]
Ok @patrikgrenfeldt @henrikstranneheim , I would like to merge this now

File: cg/apps/crunchy/crunchy.py [hunk 2]
```diff
@@ -54,108 +42,195 @@ def __init__(self, config: dict):
...
+        if not compression_obj.metadata_exists():
+            LOG.info("No metadata file found")
+            return False
+        LOG.info("SPRING metadata file found")
+
+        spring_metadata = self.get_spring_metadata(compression_obj.spring_metadata_path)
+        updated_at = self.get_file_updated_at(spring_metadata)
 
-    def fastq_to_spring(self, fastq_first: Path, fastq_second: Path, sample_id: str = ""):
+        if updated_at is None:
```
[the remark this answers is not in the document]
Comment (moonso) [reply cg/apps/crunchy/crunchy.py:136]: Yes, I have added that

File: cg/meta/compress/compress.py [hunk 3]
```diff
@@ -207,28 +121,32 @@ def clean_fastq(self, sample_id: str) -> bool:
...
-            fastq_first_hk = sample_fastq_dict[run]["fastq_first_file"]["hk_file"]
-            fastq_second_hk = sample_fastq_dict[run]["fastq_second_file"]["hk_file"]
+            LOG.info("FASTQ compression done for sample %s, run %s!!", sample_id, run_name)
 
             self.update_fastq_hk(
-                sample_id=sample_id, hk_fastq_first=fastq_first_hk, hk_fastq_second=fastq_second_hk
+                sample_id=sample_id,
+                compression_obj=compression_object,
+                hk_fastq_first=sample_fastq_dict[run_name]["hk_first"],
+                hk_fastq_second=sample_fastq_dict[run_name]["hk_first"],
```
[the remark this answers is not in the document]
Comment (moonso) [reply cg/meta/compress/compress.py:139]: 😅

File: tests/apps/crunchy/test_spring_decompression.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    crunchy_api = CrunchyAPI(crunchy_config_dict)
+    # GIVEN that the fastq paths exists
+    compression_object.fastq_first.touch()
+    compression_object.fastq_second.touch()
+    # GIVEN a existing spring file
+    compression_object.spring_path.touch()
+
+    # GIVEN a existing flag file
+    # GIVEN that the files have an updated tag
+    old_date = "2019-01-01"
```
[the remark this answers is not in the document]
Comment (moonso) [reply tests/apps/crunchy/test_spring_decompression.py:27]: Only used once but I'll have it in mind

**Comment by moonso** [comment]
Skärmavbild 2020-08-27 kl  19 58 54

Merged and deployed on hasta

**Pull request merged**